lib/cc/analyzer/engine.rb

```diff
@@ -16,7 +16,9 @@ def initialize(name, metadata, code_path, config, label)
         @label = label.to_s
       end
 
-      def run(stdout_io, stderr_io = StringIO.new)
+      def run(stdout_io, stderr_io = StringIO.new, container_log = NullContainerLog.new)
+        container_log.started(@metadata["image"])
```

Comment:
The object that would initialize this log instance wouldn't know the image name. Since we've got the `EnginesRunner` class in the middle, the original invoking doesn't know anything engine-specific unless it's passed into the life-cycle methods themselves (like this).